### PR TITLE
Add support for Jenkins CI; restructure scripts accordingly

### PR DESCRIPTION
**Pull request by jramsdale**

### CODE REVIEW

File: scripts/validate.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#!/bin/bash -e
```
Comment (chrislovecnm) [inline scripts/validate.sh]: We should be using env not bash directly. Probably missed this last review

File: scripts/validate.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+then
+  echo "ERROR - Application failed to deploy"
+  exit 1
+fi
+
+echo "Step 1 of the validation passed. App is deployed."
+
+# Grab the external IP and port of the service to confirm that demo app
+#   deployed correctly.
+EXT_IP=$(kubectl get svc "$APP_NAME" -n default \
```
Comment (chrislovecnm) [inline scripts/validate.sh]: We are not waiting for the service to get an ip? Often this takes a bit.

Comment (jramsdale) [reply scripts/validate.sh]: That's exactly what I found: https://ci.gflocks.com/job/gke-monitoring-tutorial/1/console

Will need to be changed in the other one, too.

File: scripts/validate.sh (hunk 1 above)
Comment (jramsdale) [reply scripts/validate.sh]: Gotcha. Will change here and there.

File: scripts/common.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash -e
+#! /usr/bin/env bash
```
Comment (chrislovecnm) [inline scripts/common.sh]: Yah dropped the -e. Any reason?

Comment (jramsdale) [reply scripts/common.sh]: Copied it from somewhere! Can you give me the full, correct, reference I should use? I'll normalize it everywhere.

Comment (jramsdale) [reply scripts/common.sh]: Reading up on it, I'm not sure we want to use the -e. We set error handling at the top of each script and they don't all use the same behavior. Thoughts?

File: Makefile [hunk 4]
```diff
@@ -15,8 +15,25 @@
 # Make will use bash instead of sh
 SHELL := /usr/bin/env bash
 
-# All is the first target in the file so it will get picked up when you just run 'make' on its own
-all: check_shell check_python check_golang check_terraform check_docker check_base_files check_headers check_trailing_whitespace
+ROOT := ${CURDIR}
```
Comment (techgnosis) [inline Makefile]: Is this line getting used?

File: scripts/validate.sh [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+then
+  echo "ERROR - Application failed to deploy"
+  exit 1
+fi
+
+echo "Step 1 of the validation passed. App is deployed."
+
+# Loop for up to 60 seconds waiting for service's IP address
+EXT_IP=""
+for ((i=0; i < 30 ; i++)); do
```
Comment (techgnosis) [inline scripts/validate.sh]: Two different syntaxes for looping 30 times are used. One in this loop and another syntax in the loop before it. Let's be consistent.

File: scripts/common.sh (hunk 3 above)
Comment (techgnosis) [reply scripts/common.sh]: Yeah you can't give more than one parameter in a shebang according to shellcheck. If you want error checking when using '/usr/bin/env bash' then you have to set it with a 'set'

**Comment by chrislovecnm** [comment]
@jramsdale another few tweaks from @techgnosis  for you

File: Makefile (hunk 4 above)
Comment (jramsdale) [reply Makefile]: I guess not? Removed.

File: scripts/validate.sh (hunk 5 above)
Comment (jramsdale) [reply scripts/validate.sh]: Agreed. This was a vestige of an earlier pursuit that had more than just a counter. Fixed.

**Pull request merged**